span.go

```diff
@@ -113,7 +116,14 @@ func (s *spanImpl) Log(ld opentracing.LogData) {
 		ld.Timestamp = time.Now()
 	}
 
-	s.raw.Logs = append(s.raw.Logs, ld)
+	if maxLogs := s.tracer.options.MaxLogsPerSpan; maxLogs > 0 && len(s.raw.Logs) == maxLogs {
+		// We have too many logs. Overwrite the earliest log (use Logs as a circular
+		// buffer).
+		s.raw.Logs[s.numDroppedLogs%maxLogs] = ld
```

Comment:
to make this efficient, how about treating the 2nd half of the array as a circular buffer?